app/Rules/AlbumIDListRule.php

```diff
@@ -17,13 +17,10 @@ public function passes($attribute, $value): bool
 			return false;
 		}
 		$albumIDs = explode(',', $value);
-		if (!is_array($albumIDs) || count($albumIDs) === 0) {
-			return false;
-		}
```

Comment:
I'm curious about the elimination of this test. I can imagine that `is_array` may not be needed if that's always what `explode` returns, but why are we eliminating the `count` test as well? It looks like that will change the behavior of this method for empty `$value`?

Reply:
because explode always return an album with at least one element. :)

Reply:
How do we know that? Is `$value` guaranteed to be non-empty?

Reply:
> According to the docs explode throws ValueError if the empty string '' is passed in instead of returning false since PHP 8.0.

From the doc:
> If **separator** is an empty string (""), explode() throws a [ValueError](https://www.php.net/manual/en/class.valueerror.php).

In our case the separator is `','` So no problem here.

Reply:
Ups, proper reading is an advantage. :joy:

Reply:
But there is still a change in behavior. The return value. You are right, there is no problem, because `foreach` will simply iterate over an empty array. This is fine. But previously, we returned `false` for an empty list. Now, we return `true`. However, it is necessary to return `false` such that the client gets a 422 error response. The client is not expected to post an empty list.

If we continue with an empty album list, because the empty list passed the test, the request may fail later with some unexpected error. In may cases it might even work. For example, if the client requests a ZIP archive and does not pass any album IDs, the client might get a ZIP archive which is simply empty (I haven't tested it, if this really happens). But it still makes not much sense. So, I would prefer to return `false` for empty lists, just to be an the safe side.

Reply:
```php
explode(',', "");
```
returns 
```
[""]
```
which is an array of 1 element.
Thus `!is_array` returns `false` and `count(...) === 0` also returns `false`.